src/main/java/org/araymond/joal/core/bandwith/BandwidthDispatcher.java

```diff
@@ -129,7 +131,7 @@ public void registerTorrent(final InfoHash infoHash) {
         this.lock.writeLock().lock();
         try {
             this.torrentsSeedStats.put(infoHash, new TorrentSeedStats());
-            this.speedMap.put(infoHash, new Speed(0));
+            this.speedMap.put(infoHash, new Speed(0));  // TODO: why start w/ 0 speed?
```

Comment:
`registerTorrent()` has no responsability over speed attribution. The spread attribution belongs to `recomputeSpeeds()`, it dispatch speed based on peers stats.
I prefer the torrent to start at 0, then upgraded (or not) by the `recomputeSpeeds()` rather than introducing speed logic here.

Reply:
The speed will be updated by `updateTorrentPeers()` when the torrent first announce successfully and report his peers to the BandiwidthDispatcher

Reply:
```suggestion
            this.speedMap.put(infoHash, new Speed(0));
```

Reply:
When you put it like this it makes perfect sense. I'm overthinking something here...